### PR TITLE
[ObstacleLayer] Use message_filter timeout

### PR DESCRIPTION
**Pull request by doisyg**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | https://github.com/ros-planning/navigation2/issues/2333 |
| Primary OS tested on | Ubuntu|
| Robotic platform tested on | Elodie Wyca|

---

## Description of documentation updates required from your changes
Use a timeout on the obstacle_layer message_filter. And use `transform_tolerance` to set its value since buffered message from the message_filter older than `transform_tolerance` would have been discarded later.
This effectively solves the controller_server being stuck with a very high CPU usage after few tf extrapolation errors

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

File: nav2_costmap_2d/plugins/obstacle_layer.cpp [hunk 1]
```diff
@@ -218,7 +218,7 @@ void ObstacleLayer::onInitialize()
 
       std::shared_ptr<tf2_ros::MessageFilter<sensor_msgs::msg::LaserScan>> filter(
         new tf2_ros::MessageFilter<sensor_msgs::msg::LaserScan>(
-          *sub, *tf_, global_frame_, 50, rclcpp_node_));
+          *sub, *tf_, global_frame_, 50, rclcpp_node_, tf2::durationFromSec(transform_tolerance)));
```
Comment (SteveMacenski) [inline nav2_costmap_2d/plugins/obstacle_layer.cpp:221]: You mind adding this to the other message filters? There's another in this file and 1 in AMCL. There's also one in SLAM Toolbox / STVL each if you're still users of those

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

File: nav2_costmap_2d/plugins/obstacle_layer.cpp (hunk 1 above)
Comment (SteveMacenski) [reply nav2_costmap_2d/plugins/obstacle_layer.cpp:221]: Edit: You were ahead of me :laughing:

Comment (doisyg) [reply nav2_costmap_2d/plugins/obstacle_layer.cpp:221]: already done for STVL (as yes, I am still using it), I am checking the others

Comment (SteveMacenski) [reply nav2_costmap_2d/plugins/obstacle_layer.cpp:221]: there's 2 of these in obstacle layer; one for PC2 and another for laser scans

Comment (doisyg) [reply nav2_costmap_2d/plugins/obstacle_layer.cpp:221]: Can't believe I missed the PC2 one, done now. Troubleshooting this problem really got the best of me.
However, I don't use SLAM toolbox and I am not familiar with its code, so I will let you fix it there if that's fine.

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
Waiting on CI

**Pull request merged**